### PR TITLE
Support remapping annotations in classes, methods and fields

### PR DESCRIPTION
**Pull request by NZSmartie**
Fixes https://github.com/FabricMC/tiny-remapper/issues/5

To the extent possible under law, Roman Vaughan has waived all copyright and related or neighboring rights to this work.

Signed-off-by: NZSmartie <nzsmartie@gmail.com>

### CODE REVIEW

File: src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java [hunk 1]
```diff
@@ -143,4 +169,37 @@ private static boolean isJavaLambdaMetafactory(Handle bsm) {
...
+		public AsmAnnotationRemapper(String baseDesc, AnnotationVisitor annotationVisitor, Remapper remapper) {
+			super(annotationVisitor, remapper);
+			annotationClass = Type.getType(baseDesc).getInternalName();
+		}
+
+		@Override
+		public void visit(String name, Object value) {
+			String desc = "()"+Type.getDescriptor(value.getClass());
+			if(remapper instanceof AsmRemapper) {
+				System.err.printf("remapping annotation %s -> %s\n", name, remapper.mapMethodName(annotationClass, name, desc));
```
Comment (JamiesWhiteShirt) [inline src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java]: Is this supposed to be here?

File: src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java [hunk 2]
```diff
@@ -143,4 +169,37 @@ private static boolean isJavaLambdaMetafactory(Handle bsm) {
 					&& !bsm.isInterface();
 		}
 	}
+
+	@Override
+	public AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
+		return createAsnAnnotationRemapper(descriptor, super.visitAnnotation(descriptor, visible), remapper);
+	}
+
+	public static AnnotationRemapper createAsnAnnotationRemapper(String desc, AnnotationVisitor annotationVisitor, Remapper remapper) {
```
Comment (JamiesWhiteShirt) [inline src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java]: Did you mean `Asm`?

Comment (NZSmartie) [reply src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java]: Yes, thank you!

File: src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java (hunk 1 above)
Comment (NZSmartie) [reply src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java]: Nope, was for debugging only

**Pull request merged**